src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1873,6 +1873,109 @@ private static bool TryGetCommandInfoList(PowerShell ps, HashSet<string> command
             return true;
         }
 
+        internal static OutputRendering OutputRenderingSetting = OutputRendering.Automatic;
+
+        internal static bool ShouldOutputPlainText(bool isHost, bool? supportsVirtualTerminal)
+        {
+            var outputRendering = OutputRendering.Ansi;
+
+            if (ExperimentalFeature.IsEnabled("PSAnsiRendering"))
+            {
+                if (supportsVirtualTerminal != false)
+                {
+                    switch (OutputRenderingSetting)
+                    {
+                        case OutputRendering.Automatic:
+                            outputRendering = OutputRendering.Ansi;
+                            break;
+                        case OutputRendering.Host:
+                            outputRendering = isHost ? OutputRendering.Ansi : OutputRendering.PlainText;
+                            break;
+                        default:
+                            outputRendering = OutputRenderingSetting;
+                            break;
+                    }
+                }
+            }
+
+            return outputRendering == OutputRendering.PlainText;
+        }
+
+        internal static string GetOutputString(string s, bool isHost, bool? supportsVirtualTerminal = null, bool isOutputRedirected = false)
```

Comment:
This API may need to be public, so that other host implementations can strip off ANSI sequences the same way PowerShell does.

Reply:
When thinking of a public API, it's better to not have default arguments, so maybe have an internal overload  for `GetOutputString(string s, bool isHost)`. Also, do we need to use `bool?`, by looking at the use of this parameter, it seems `bool` should be sufficient.

Reply:
Will add overloads and make public

Reply:
So these static methods are currently on the Utils internal class.  Perhaps we should wait on making a public API until after there is a need?

Reply:
I think all existing powershell hosts will need it when moving to PS 7.2, otherwise, their users will find `$PSStyle.OutputRendering` doesn't take effect. But I'm fine to expose the API in a separate PR. Let's open an issue to track it.

Reply:
There is a need, for sure! I have developed my own versions of this as there has been no API.
I even think we should consider an API that doesn't just expose the implementation details but to think about what devs would like to do with it.
Otherwise, everyone will have to invent that wheel over and over again.

Another opportunity if you use methods is that you can determine if ANSI-output isn't enabled and then just bypass the decorations.

Reply:
Open the issue https://github.com/PowerShell/PowerShell/issues/14387 to track this work.